CorsixTH/CMakeLists.txt

```diff
@@ -99,6 +58,23 @@ if(UNIX AND NOT APPLE)
   set_target_properties(CorsixTH PROPERTIES OUTPUT_NAME corsix-th)
 endif()
 
+# Generate source files list
+# Note: Done after generating targets
+add_subdirectory(${PROJECT_SOURCE_DIR}/Src)
+add_subdirectory(${PROJECT_SOURCE_DIR}/SrcUnshared)
+
+# Manually add common files - as different targets could include the file 
+# so we can't just do it in in the folder like above
+target_sources(CorsixTH_lib
+  PRIVATE
+  ${CMAKE_SOURCE_DIR}/common/rnc.cpp
+  ${CMAKE_SOURCE_DIR}/common/rnc.h
+)
+
+# Ensure config.h is picked up by cmake - moving this into subdir cmake files will
+# prevent it applying to the CorsixTH project
+include_directories(${CMAKE_BINARY_DIR}/CorsixTH/Src/)
```

Comment:
I think what you are looking for is `target_include_directories` https://cmake.org/cmake/help/v3.0/command/target_include_directories.html